The-Compiler/vscode-python-tox
# Provide a preview feature for tox.ini files

## Addressed Changed

According to issue #19:

* Show previews via **hover messages** for environment variables in `tox.ini` files for `passenv` and `setenv` variables.
* Variables could be assigned directly in a section (environment) or in the inherited section `testenv` which can be reused by all other sections (environments).
* `passenv` variables are read from the current environment.
* `setenv` variables are assigned either directly inside the tox.ini file or indirectly referenced from an external `.env` file.

## Changes

* Added class `src/environment_variables_service.ts` which contains the logic to provide hover messages with preview data for tox environment variables.
* Added unit tests and sample files (in `src/test/examples/envvars/`) to test new features for previewing environment variables in `tox.ini` files.
* Added register command for VS Code hover integration to `src/extension.ts`.
* Added activation event in `package.json` for extended language support in `ini` files.

## How to test

* run unit tests via `npm test` from terminal or run `Extension Tests` in VS Code
* run extension, open a `tox.ini` file, and hover with the mouse over an environment variable. You will see a preview of the value of the variable.

![image](https://user-images.githubusercontent.com/5375651/165092503-fc9b86fb-4361-4df6-a776-2236d6de779c.png)
